contracts/yGift/yGift.sol

```diff
@@ -50,8 +50,8 @@ contract yGift is ERC721("yearn Gift NFT", "yGIFT") {
 		uint _start,
 		uint _duration)
 	external {
-		uint _id = gifts.length;
-		gifts.push(Gift({
+		uint _id = totalSupply();
```

Comment:
Checked that you don't need to update totalSupply manually, it's read dynamically by the OZ library.
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/883116e4afa59a1155fcd4a4c394819e636794e6/contracts/utils/EnumerableMap.sol#L126